### PR TITLE
feat:Update to Kochava 4.2.1

### PR DESCRIPTION
**Pull request by BrandonStalnaker**
## Summary
 - Updated to Kochava 4.2.1

 ## Testing Plan
 - Tested locally 

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-4963

### CODE REVIEW

File: src/main/kotlin/com/mparticle/kits/KochavaKit.kt [hunk 1]
```diff
@@ -26,27 +21,41 @@ class KochavaKit : KitIntegration(), AttributeListener {
...
         }
+        Tracker.getInstance().setLogLevel(logLevel)
+
+        Tracker.getInstance().setAppLimitAdTracking(java.lang.Boolean.parseBoolean(getSettings()[LIMIT_ADD_TRACKING]))
         val configuration = getSettings()[APP_ID]?.let {
-            Tracker.Configuration(context.applicationContext)
-                .setLogLevel(logLevel)
-                .setAppGuid(it)
-                .setAppLimitAdTracking(java.lang.Boolean.parseBoolean(getSettings()[LIMIT_ADD_TRACKING]))
+            Tracker.getInstance().startWithAppGuid(context.applicationContext, it)
```
Comment (markvdouw) [inline src/main/kotlin/com/mparticle/kits/KochavaKit.kt]: This returns a configuration? Based on the name of the function "startWithAppGuid" doesnt seems to..
If it doesn't.. line 34 wont be true and none of the code will be triggered.

File: src/main/kotlin/com/mparticle/kits/KochavaKit.kt [hunk 2]
```diff
@@ -58,24 +67,18 @@ class KochavaKit : KitIntegration(), AttributeListener {
     override fun supportsAttributeLists(): Boolean = true
     override fun setAllUserAttributes(map: Map<String, String>, map1: Map<String, List<String>>) {}
     override fun removeUserAttribute(key: String) {}
-    override fun setInstallReferrer(intent: Intent) {
-        ReferralReceiver().onReceive(context, intent)
-    }
+    override fun setInstallReferrer(intent: Intent) {}
```
Comment (markvdouw) [inline src/main/kotlin/com/mparticle/kits/KochavaKit.kt:70]: Should we propagate to the ReferralReceiver?

File: src/main/kotlin/com/mparticle/kits/KochavaKit.kt [hunk 3]
```diff
@@ -58,24 +67,18 @@ class KochavaKit : KitIntegration(), AttributeListener {
...
 
     override fun setUserIdentity(identityType: IdentityType, id: String) {
         if (identityType == IdentityType.CustomerId) {
```
Comment (markvdouw) [inline src/main/kotlin/com/mparticle/kits/KochavaKit.kt]: Shouldn't all this be simplified? Lines 77 and 81 are exactly the same

**Review by markvdouw — CHANGES_REQUESTED** [review]
Shouldn't we add some test about the new changes?

File: src/main/kotlin/com/mparticle/kits/KochavaKit.kt (hunk 1 above)
Comment (BrandonStalnaker) [reply src/main/kotlin/com/mparticle/kits/KochavaKit.kt]: configuration is being set by 'getSettings()[APP_ID]?.let' here right. Maybe let works way differently here than it does in Swift?

File: src/main/kotlin/com/mparticle/kits/KochavaKit.kt (hunk 2 above)
Comment (BrandonStalnaker) [reply src/main/kotlin/com/mparticle/kits/KochavaKit.kt:70]: In their new sdk I couldn't find anything comparable to ReferralReceiver or any way ingest intent. I believe the functionality has been removed but let me know if you see different https://support.kochava.com/sdk-integration/android-sdk-integration/android-using-the-sdk/

File: src/main/kotlin/com/mparticle/kits/KochavaKit.kt (hunk 1 above)
Comment (markvdouw) [reply src/main/kotlin/com/mparticle/kits/KochavaKit.kt]: I was just saying that the result of the let block is being assigned to the immutable configuration variable. So my question was what is being assigned? Does "startWithAppGuid" returns a configuration?

Comment (BrandonStalnaker) [reply src/main/kotlin/com/mparticle/kits/KochavaKit.kt]: Gotcha, I didn't realize the block assigned to the original value. Fixed

**Review by markvdouw — APPROVED** [review]
LGTM

**Pull request merged**